uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -250,25 +274,45 @@ define([
         vm.error = undefined;
       };
 
-      (function () {
-        // Create an empty leave request
-        vm.leaveRequest = LeaveRequestInstance.init({
-          contact_id: contactId //resolved from directive
-        }, false);
+      /**
+       * initializes the controller on loading the dialog
+       */
+      (function initController() {
+        if (directiveOptions.leaveRequest) {
+          //manager is responding to leave request
+          vm.uiOptions.isManager = directiveOptions.leaveRequest.contact_id != directiveOptions.contactId;
+          //owner is editing popup
+          vm.uiOptions.isOwner = directiveOptions.leaveRequest.contact_id == directiveOptions.contactId;
+        }
+
+        if (canEdit()) {
+          //get a clone
+          var clone = _.cloneDeep(directiveOptions.leaveRequest);
+          vm.leaveRequest = LeaveRequestInstance.init(clone, true);
```

Comment:
1. There is no need to re-init the leave request, you are already cloning an initialized leave request instance
2. the second parameter for `init()` is for when you are initializing the model instance with data coming straight from the API, which is not something that is happening here